src/test/java/seedu/address/model/attendance/AttendanceRecordListTest.java

```diff
@@ -0,0 +1,231 @@
+import static seedu.address.logic.commands.CommandTestUtil.VALID_NUMBER_OF_OCCURRENCES_7;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalAttendanceRecord.RECORD_ALICE_80;
+import static seedu.address.testutil.TypicalAttendanceRecord.RECORD_EMPTY;
+import static seedu.address.testutil.TypicalAttendanceRecord.getTypicalAttendanceRecord;
+import static seedu.address.testutil.TypicalStudent.ALICE;
+import static seedu.address.testutil.TypicalStudent.BENSON;
+import static seedu.address.testutil.TypicalStudent.CARL;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.attendance.exceptions.AttendanceNotFoundException;
+import seedu.address.model.attendance.exceptions.DuplicateAttendanceException;
+import seedu.address.model.attendance.exceptions.InvalidWeekException;
+import seedu.address.model.lesson.NumberOfOccurrences;
+
+public class AttendanceRecordListTest {
+
+    private static final NumberOfOccurrences numberOfOccurrences = new NumberOfOccurrences(
+            VALID_NUMBER_OF_OCCURRENCES_7);
+    private static final AttendanceRecordList recordList = new AttendanceRecordList(
+            numberOfOccurrences.getNumberOfOccurrences());
+    private static final Week VALID_WEEK = new Week(1);
+    private static final Attendance VALID_ATTENDANCE = new Attendance(10);
+
+    @Test
+    public void constructor_emptyAttendanceRecordMap() {
+        assertTrue(recordList.getAttendanceRecordList().size() == 7);
+        for (AttendanceRecord record: recordList.getAttendanceRecordList()) {
+            assertTrue(record.equals(RECORD_EMPTY));
+        }
+    }
+
+    @Test
+    public void getAttendance_nullStudent_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> recordList.getAttendance(null, VALID_WEEK));
+    }
+
+    @Test
+    public void getAttendance_nullWeek_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> recordList.getAttendance(ALICE, null));
+    }
+
+    @Test
+    public void getAttendance_invalidWeek_throwsInvalidWeekException() {
+        assertThrows(InvalidWeekException.class, () -> recordList.getAttendance(
+                ALICE, new Week(VALID_NUMBER_OF_OCCURRENCES_7 + 1)));
+    }
+
+    @Test
+    public void getAttendance_nonExistingStudent_throwsAttendanceNotFoundException() {
+        assertThrows(AttendanceNotFoundException.class, () -> recordList.getAttendance(ALICE, VALID_WEEK));
+    }
+
+    @Test
+    public void getAttendance_existingStudent_success() {
+        AttendanceRecordList attendanceRecordList = createAliceRecordList();
+        Attendance attendance = attendanceRecordList.getAttendance(ALICE, new Week(1));
+        assertEquals(new Attendance(VALID_ATTENDANCE_80), attendance);
+    }
+
+    @Test
+    public void addAttendance_nullStudent_throwsNullPointerException() {
```

Comment:
I agree with Dexter on this. We can probably extract the CRUD operations to Logic instead of clustering everything in AttendanceRecordListTest.java